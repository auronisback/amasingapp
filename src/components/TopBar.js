import React, {Component} from 'react';
import {library} from '@fortawesome/fontawesome-svg-core';
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faSearch, faUndo} from "@fortawesome/free-solid-svg-icons";


export class TopBar extends Component {

  /**
   * Component's state.
   * */
  state = {
    streetName: ''
  };

  constructor(props) {
    super(props);
    // Initializing font-awesome icons
    library.add(faSearch, faUndo);
  }

  render() {
    return (
        <header className={'container fluid TopBar'}>
          <div className={'row'}>
            <div className={'col-6'}>
              <h1>aMasIngApp</h1>
            </div>
            <div className={'col-6'}>
              <ul>
                <li>
                  <button className={'btn btn-primary ResetButton'}
                          onClick={this.props.onReset}>
                    <FontAwesomeIcon icon={'undo'} />
                  </button>
                </li>
                <li>
                  <div className={'input-group'}>
                    <input type={'text'} className={'form-control'} placeholder={'Search by street'}
                           onChange={this.onSearchInputChanged}
                           value={this.state.streetName}
                           onKeyPress={this.onInputKeyPressed}
                    />
                    <button className={'input-group-append btn btn-primary SearchButton'}
                            onClick={this.onSearch}
                    >
                      <FontAwesomeIcon icon={"search"}/>
                    </button>
                  </div>
                </li>
              </ul>
            </div>
          </div>
        </header>
    );
  }

  /**
   * Action to be performed when the search button is pressed. The action
   * will be performed if the input has at least two characters.
   * */
  onSearch = () => {
    if (this.props.onSearch && this.state.streetName.length > 2)
      this.props.onSearch(this.state.streetName); // Taken from component's state
  };

  /**
   * Handler for changing the street name input.
   * @param {Event} evt The generated event
   * */
  onSearchInputChanged = (evt) => {
    this.setState({streetName: evt.target.value});
  };

  /**
   * Handler for the pressure of a key in the input. If the pressed key
   * is ENTER, then it will fire the search.
   * @param {Event} evt The generated event
   * */
  onInputKeyPressed = (evt) => {
    if(evt.key === 'Enter')
      this.onSearch();
  };
}