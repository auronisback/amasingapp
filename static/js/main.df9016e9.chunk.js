(window.webpackJsonp=window.webpackJsonp||[]).push([[0],{167:function(e,t,a){e.exports=a(398)},195:function(e,t,a){},398:function(e,t,a){"use strict";a.r(t);var n=a(1),r=a.n(n),i=a(8),s=a.n(i),o=a(9),c=a(10),l=a(12),p=a(11),u=a(13),h=a(99),m=a.n(h),d=a(100),g=a.n(d),f=(a(195),a(34)),E=a(37),v=a(38),b=function(e){function t(e){var a;return Object(o.a)(this,t),(a=Object(l.a)(this,Object(p.a)(t).call(this,e))).state={streetName:""},a.onSearch=function(){a.props.onSearch&&a.state.streetName.length>2&&a.props.onSearch(a.state.streetName)},a.onSearchInputChanged=function(e){a.setState({streetName:e.target.value})},a.onInputKeyPressed=function(e){"Enter"===e.key&&a.onSearch()},f.b.add(v.b,v.d),a}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){return r.a.createElement("header",{className:"container fluid TopBar"},r.a.createElement("div",{className:"row"},r.a.createElement("div",{className:"col-6"},r.a.createElement("h1",null,"aMasIngApp")),r.a.createElement("div",{className:"col-6"},r.a.createElement("ul",null,r.a.createElement("li",null,r.a.createElement("button",{className:"btn btn-primary",onClick:this.props.onReset},r.a.createElement(E.a,{icon:"undo"}))),r.a.createElement("li",null,r.a.createElement("div",{className:"input-group"},r.a.createElement("input",{type:"text",className:"form-control",placeholder:"Search by street",onChange:this.onSearchInputChanged,value:this.state.streetName,onKeyPress:this.onInputKeyPressed}),r.a.createElement("button",{className:"input-group-append btn btn-primary SearchButton",onClick:this.onSearch},r.a.createElement(E.a,{icon:"search"}))))))))}}]),t}(n.Component),k=a(17),O=Object(k.withScriptjs)(Object(k.withGoogleMap)(function(e){return r.a.createElement(k.GoogleMap,{center:e.center,defaultZoom:e.zoom,onClick:e.onMapClick},e.children)})),y=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,r=new Array(n),i=0;i<n;i++)r[i]=arguments[i];return(a=Object(l.a)(this,(e=Object(p.a)(t)).call.apply(e,[this].concat(r)))).onMapClick=function(e){a.props.onDestinationSelection&&a.props.onDestinationSelection(e.latLng.lat(),e.latLng.lng())},a}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){var e=this.props,t=e.center,a=e.zoom,n=e.gMapsApiKey;return r.a.createElement("div",null,r.a.createElement(O,{googleMapURL:"https://maps.googleapis.com/maps/api/js?key="+n+"&v=3.exp&libraries=geometry,drawing,places",loadingElement:r.a.createElement("div",{style:{height:"100px"}}),containerElement:r.a.createElement("div",{className:"MapPanel"}),mapElement:r.a.createElement("div",{style:{height:"100%"}}),center:t,zoom:a,onMapClick:this.onMapClick},this.props.children))}}]),t}(n.Component),S=a(166),C=(a(397),function(e){function t(e){var a;return Object(o.a)(this,t),(a=Object(l.a)(this,Object(p.a)(t).call(this,e))).onDowChanged=function(e){a.setState({dow:e.target.value})},a.onTimeChanged=function(e){var n=t.getTimeFromString(e.target.value);a.props.onTimeChanged({h:n.getHours(),m:n.getMinutes()})},a.onDistanceChanged=function(e){a.props.onDistanceChanged(e.target.value)},a.onSliderChanged=function(e){a.props.onWalkWeightChanged(e)},a.onFormSubmitted=function(e){e.nativeEvent.preventDefault(),a.props.onSubmit&&a.props.onSubmit()},f.b.add(v.e,v.a,v.c),a}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){return r.a.createElement("form",{onSubmit:this.onFormSubmitted},r.a.createElement("div",{className:"SearchPanel form-group"},r.a.createElement("select",{className:"form-control",value:this.props.dow,onChange:this.onDowChanged},r.a.createElement("option",{value:0},"Sunday"),r.a.createElement("option",{value:1},"Monday"),r.a.createElement("option",{value:2},"Tuesday"),r.a.createElement("option",{value:3},"Wednesday"),r.a.createElement("option",{value:4},"Thursday"),r.a.createElement("option",{value:5},"Friday"),r.a.createElement("option",{value:6},"Saturday")),r.a.createElement("input",{type:"time",className:"form-control",value:this.props.time.h+":"+(""+this.props.time.m).padStart(2,"0"),onChange:this.onTimeChanged}),r.a.createElement("input",{type:"number",placeholder:"distance (m)",className:"form-control",value:0===this.props.distance?"":this.props.distance,onChange:this.onDistanceChanged,min:0,step:50}),r.a.createElement("div",{className:"WalkingSlider"},r.a.createElement("div",{className:"IconWrapper"},r.a.createElement(E.a,{icon:"blind"})),r.a.createElement(S.a,{className:"Slider",onChange:this.onSliderChanged,value:this.props.walkWeight}),r.a.createElement("div",{className:"IconWrapper"},r.a.createElement(E.a,{icon:"walking"}))),r.a.createElement("button",{className:"btn btn-primary btn-sm",disabled:!!this.props.isLoading},this.props.isLoading?r.a.createElement(E.a,{icon:"spinner",pulse:!0}):"Search")))}}],[{key:"getTimeFromString",value:function(e){var t=e.split(":");return(e=new Date).setHours(t[0]),e.setMinutes(t[1]),e}}]),t}(n.Component)),w=100,j=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,r=new Array(n),i=0;i<n;i++)r[i]=arguments[i];return(a=Object(l.a)(this,(e=Object(p.a)(t)).call.apply(e,[this].concat(r)))).state={distance:w,dow:(new Date).getDay(),time:{h:(new Date).getHours(),m:(new Date).getMinutes()},walkWeight:50},a.onDowUpdate=function(e){a.setState({dow:e})},a.onTimeUpdate=function(e){a.setState({time:e})},a.onDistanceUpdate=function(e){a.setState({distance:+e>0?+e:0})},a.onWalkWeightUpdate=function(e){a.setState({walkWeight:e})},a.onSubmit=function(){a.props.onSearch&&a.props.onSearch(a.state)},a}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){if(!this.props.destination)return null;var e,a=this.props.searchVisible?r.a.createElement(k.OverlayView,{position:this.props.destination,mapPaneName:k.OverlayView.OVERLAY_MOUSE_TARGET,getPixelPositionOffset:(e=t.markerPixelOffset,function(t,a){return{x:-t/2+e.x,y:-a/2+e.y}}),options:{enableEventPropagation:!1}},r.a.createElement(C,{onSubmit:this.onSubmit,className:"SearchOpened",isLoading:this.props.isLoading,dow:this.state.dow,distance:this.state.distance,time:this.state.time,walkWeight:this.state.walkWeight,onDowChanged:this.onDowUpdate,onTimeChanged:this.onTimeUpdate,onDistanceChanged:this.onDistanceUpdate,onWalkWeightChanged:this.onWalkWeightUpdate})):null;return r.a.createElement(k.Marker,{position:this.props.destination},a,r.a.createElement(k.Circle,{center:this.props.destination,radius:this.state.distance,options:{fillColor:t.CIRCLE_FILL_COLOR,strokeWeight:t.CIRCLE_STROKE_WEIGHT}}))}}]),t}(n.Component);j.CIRCLE_FILL_COLOR="#F00",j.CIRCLE_STROKE_WEIGHT=0,j.markerPixelOffset={x:0,y:-32};var R="Server returned status code ZERO_RESULTS",_=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,r=new Array(n),i=0;i<n;i++)r[i]=arguments[i];return(a=Object(l.a)(this,(e=Object(p.a)(t)).call.apply(e,[this].concat(r)))).getMessage=function(){switch(a.props.message){case R:return"The search produced no results.";default:return a.props.message}},a}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){var e=this.getMessage(),t="alert-"+this.props.type;return r.a.createElement("div",{className:"alert "+t+" ErrorDialog",onClick:this.props.onClose},r.a.createElement("h4",{className:"alert-heading"},"Warning!"),r.a.createElement("p",null,e),r.a.createElement("span",{className:"CloseError"},"Click to close"))}}]),t}(n.Component);_.TYPE_MESSAGE="info",_.TYPE_WARNING="warning",_.TYPE_ERROR="danger";var N=function(e){function t(){return Object(o.a)(this,t),Object(l.a)(this,Object(p.a)(t).apply(this,arguments))}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){return r.a.createElement(k.Polyline,{path:[this.props.start,this.props.end],options:{strokeColor:this.props.color,strokeOpacity:this.props.opacity}})}}]),t}(n.Component),T=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,i=new Array(n),s=0;s<n;s++)i[s]=arguments[s];return(a=Object(l.a)(this,(e=Object(p.a)(t)).call.apply(e,[this].concat(i))))._buildAvailableParks=function(){var e=[];return a.props.parks.forEach(function(a){e.push(r.a.createElement(N,{key:a.id,start:a.start,end:a.end,color:t.STROKE_COLOR,opacity:t.STROKE_OPACITY}))}),e},a}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){return r.a.createElement("div",null,this._buildAvailableParks())}}]),t}(n.Component);T.STROKE_COLOR="#777",T.STROKE_OPACITY=.5;var P=function(e){function t(){return Object(o.a)(this,t),Object(l.a)(this,Object(p.a)(t).apply(this,arguments))}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){return r.a.createElement("div",{className:"ParkInfoWindow container"},r.a.createElement("div",{className:"row"},"Occupancy Rate: ",(100*this.props.occRate).toFixed(0),"%"),r.a.createElement("div",{className:"row"},"Distance: ",t._distanceToString(this.props.distance)),r.a.createElement("div",{className:"row"},"Park Number: ",this.props.parkNumber),r.a.createElement("div",{className:"row"},"Type: ",this.props.streetType),r.a.createElement("div",{className:"row"},"Had measure: ",this.props.haveMes?"Yes":"No"))}}],[{key:"_distanceToString",value:function(e){var t=(+e).toFixed(0);return t<1e3?t+"m":(t/1e3).toFixed(1)+"km"}}]),t}(n.Component),M=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,r=new Array(n),i=0;i<n;i++)r[i]=arguments[i];return(a=Object(l.a)(this,(e=Object(p.a)(t)).call.apply(e,[this].concat(r)))).state={infoWindowShown:!1},a.toggleOpen=function(){a.setState({infoWindowShown:!a.state.infoWindowShown})},a.handleClose=function(){a.setState({infoWindowShown:!1})},a.getPosition=function(){return{lat:(a.props.parking.start.lat+a.props.parking.end.lat)/2,lng:(a.props.parking.start.lng+a.props.parking.end.lng)/2}},a}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){return r.a.createElement(k.Marker,{position:this.getPosition(),onClick:this.toggleOpen,label:""+this.props.label},this.state.infoWindowShown&&r.a.createElement(k.InfoWindow,{onCloseClick:this.handleClose},r.a.createElement(P,{occRate:this.props.parking.occupancy,distance:this.props.parking.distance,parkNumber:this.props.parking.number,streetType:this.props.parking.streetType,haveMes:this.props.parking.haveMes})),r.a.createElement(N,{start:this.props.parking.start,end:this.props.parking.end,color:t.STROKE_COLOR,opacity:t.STROKE_OPACITY}))}}]),t}(n.Component);M.STROKE_COLOR="#F00",M.STROKE_OPACITY=1;var A=function(e){function t(){var e,a;Object(o.a)(this,t);for(var n=arguments.length,i=new Array(n),s=0;s<n;s++)i[s]=arguments[s];return(a=Object(l.a)(this,(e=Object(p.a)(t)).call.apply(e,[this].concat(i))))._buildSuggestedParks=function(){var e=[];return a.props.parks.forEach(function(t,a){e.push(r.a.createElement(M,{key:t.id,parking:t,label:a+1}))}),e},a}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){return r.a.createElement("div",null,this._buildSuggestedParks())}}]),t}(n.Component),D=function(e){function t(e){var a;return Object(o.a)(this,t),(a=Object(l.a)(this,Object(p.a)(t).call(this,e))).state={destination:null,parks:[],availableParks:[],searchVisible:!0,searching:!1},a.setDestination=function(e,t){null==a.state.destination&&a.setState({destination:{lat:e,lng:t}})},a.reset=function(){a.setState({destination:null,parks:[],searchVisible:!0})},a.searchStreetByName=function(e){m.a.fromAddress(e+" San Francisco").then(function(e){var t=e.results[0].geometry.location,n=t.lat,r=t.lng;a.setState({destination:{lat:n,lng:r}})},function(e){a._showMessageDialog(e.message,_.TYPE_ERROR)})},a.onDialogClosed=function(){a.setState({error:null})},a.searchParks=function(e){a.setState({searching:!0}),e.destination=a.state.destination,a._sendRequest(e)},a._loadAllParks=function(){g.a.get(t.configuration.allParksEndpoint).then(function(e){a._parseAvailableParkResponse(e.data)},function(e){a._showMessageDialog(e.message,_.TYPE_ERROR)})},a._parseAvailableParkResponse=function(e){var t=[];e.forEach(function(e){t.push({id:e.id,start:{lat:e.line.coordinates[0][1],lng:e.line.coordinates[0][0]},end:{lat:e.line.coordinates[1][1],lng:e.line.coordinates[1][0]}})}),a.setState({availableParks:t})},a._parseMeasurementResponse=function(e){var n=[];e.forEach(function(e){n.push(a._getParkData(e))}),0===n.length?a._showMessageDialog(t.parksNotFoundMessage,_.TYPE_MESSAGE):a.setState({parks:n,searchVisible:!1})},a._showMessageDialog=function(e){var t=arguments.length>1&&void 0!==arguments[1]?arguments[1]:_.TYPE_ERROR;a.setState({error:{message:e,type:t}})},m.a.setApiKey(t.configuration.gMapsApiKey),a._loadAllParks(),a}return Object(u.a)(t,e),Object(c.a)(t,[{key:"render",value:function(){var e=null;return this.state.error&&(e=r.a.createElement(_,{message:this.state.error.message,type:this.state.error.type,onClose:this.onDialogClosed})),r.a.createElement("div",{className:"App"},r.a.createElement(b,{onReset:this.reset,onSearch:this.searchStreetByName}),r.a.createElement(y,{gMapsApiKey:t.configuration.gMapsApiKey,center:this.state.destination?this.state.destination:t.configuration.defaultMapCenter,zoom:t.configuration.mapZoom,onDestinationSelection:this.setDestination},r.a.createElement(T,{parks:this.state.availableParks}),r.a.createElement(j,{destination:this.state.destination,searchVisible:this.state.searchVisible,onSearch:this.searchParks,isLoading:this.state.searching}),r.a.createElement(A,{parks:this.state.parks})),e)}},{key:"_sendRequest",value:function(e){var a=this;g.a.post(t.configuration.measurementsEndpoint,{lat:this.state.destination.lat,lon:this.state.destination.lng,maxDistance:e.distance,dow:e.dow,hour:e.time.h,minute:e.time.m,walkWeight:1-e.walkWeight/100}).then(function(e){a._parseMeasurementResponse(e.data)},function(e){a._showMessageDialog(e.message,_.TYPE_ERROR)}).finally(function(){a.setState({searching:!1})})}},{key:"_getParkData",value:function(e){return{id:e.parkAgent.id,start:{lat:e.parkAgent.line.coordinates[0][1],lng:e.parkAgent.line.coordinates[0][0]},end:{lat:e.parkAgent.line.coordinates[1][1],lng:e.parkAgent.line.coordinates[1][0]},streetType:e.parkAgent.highway,number:e.parkAgent.parkingSpaceNumber,haveMes:e.parkAgent.haveMes,distance:e.distance,occupancy:e.meanOccRate}}}]),t}(n.Component);D.parksNotFoundMessage="No parking lots found with given criteria.",D.configuration={gMapsApiKey:"AIzaSyCra5AQ9_m4TUQRk-iVIBRKoIQvNyRsc2E",defaultMapCenter:{lat:37.773972,lng:-122.431297},mapZoom:13,measurementsEndpoint:"http://18.184.94.248:9090/measurements",allParksEndpoint:"http://18.184.94.248:9090/getParks"};var W=D;Boolean("localhost"===window.location.hostname||"[::1]"===window.location.hostname||window.location.hostname.match(/^127(?:\.(?:25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}$/));s.a.render(r.a.createElement(W,null),document.getElementById("root")),"serviceWorker"in navigator&&navigator.serviceWorker.ready.then(function(e){e.unregister()})}},[[167,1,2]]]);
//# sourceMappingURL=main.df9016e9.chunk.js.map