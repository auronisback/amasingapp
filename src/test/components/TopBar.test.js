import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, {shallow} from 'enzyme';
import {TopBar} from '../../components/TopBar.js';

// Configuring adapter
Enzyme.configure({adapter: new Adapter()});

/**
 * Top bar testings.
 * */
describe('<TopBar>', () => {
  it('Should call the search handler when searching a street', () => {
    // Mock search function
    const searchFn = jest.fn((streetname) => { });
    // Creating the top bar component
    const topBar = shallow(<TopBar
        onSearch={searchFn}
    />);
    expect(topBar).toMatchSnapshot();
    // Writing something in the input
    const streetInput = topBar.find('input');
    expect(streetInput).toHaveLength(1);
    streetInput.simulate('change', {target: {value: '1st st'}});
    // Clicking the button
    const searchButton = topBar.find('button.SearchButton');
    expect(searchButton).toHaveLength(1);
    searchButton.simulate('click');
    expect(searchFn).toBeCalledWith('1st st');
  });

  it('Should call the handler when reset button clicked', () => {
    // Mock reset function
    const resetFn = jest.fn(() => { });
    // Creating the component
    const topBar = shallow(<TopBar onReset={resetFn}/>);
    expect(topBar).toMatchSnapshot();
    // Clicking the reset button
    const resetButton = topBar.find('button.ResetButton');
    expect(resetButton).toHaveLength(1);
    resetButton.simulate('click');
    expect(resetFn).toBeCalled();
  });
});
