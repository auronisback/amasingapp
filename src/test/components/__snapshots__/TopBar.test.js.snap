// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopBar> Should call the handler when reset button clicked 1`] = `
<header
  className="container fluid TopBar"
>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <h1>
        aMasIngApp
      </h1>
    </div>
    <div
      className="col-6"
    >
      <ul>
        <li>
          <button
            className="btn btn-primary ResetButton"
            onClick={[MockFunction]}
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="undo"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
        </li>
        <li>
          <div
            className="input-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="Search by street"
              type="text"
              value=""
            />
            <button
              className="input-group-append btn btn-primary SearchButton"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="search"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
</header>
`;

exports[`<TopBar> Should call the search handler when searching a street 1`] = `
<header
  className="container fluid TopBar"
>
  <div
    className="row"
  >
    <div
      className="col-6"
    >
      <h1>
        aMasIngApp
      </h1>
    </div>
    <div
      className="col-6"
    >
      <ul>
        <li>
          <button
            className="btn btn-primary ResetButton"
          >
            <FontAwesomeIcon
              border={false}
              className=""
              fixedWidth={false}
              flip={null}
              icon="undo"
              inverse={false}
              listItem={false}
              mask={null}
              pull={null}
              pulse={false}
              rotation={null}
              size={null}
              spin={false}
              symbol={false}
              title=""
              transform={null}
            />
          </button>
        </li>
        <li>
          <div
            className="input-group"
          >
            <input
              className="form-control"
              onChange={[Function]}
              onKeyPress={[Function]}
              placeholder="Search by street"
              type="text"
              value=""
            />
            <button
              className="input-group-append btn btn-primary SearchButton"
              onClick={[Function]}
            >
              <FontAwesomeIcon
                border={false}
                className=""
                fixedWidth={false}
                flip={null}
                icon="search"
                inverse={false}
                listItem={false}
                mask={null}
                pull={null}
                pulse={false}
                rotation={null}
                size={null}
                spin={false}
                symbol={false}
                title=""
                transform={null}
              />
            </button>
          </div>
        </li>
      </ul>
    </div>
  </div>
</header>
`;
